### ISSUE TITLE
Add a function to combine genotype networks

### ISSUE THREAD

**Issue by pharos916** [ericmjl/genotype-network#41]
With the possibility of multiple data sets in the future, it might be useful to implement a function to combine genotype networks. The older version of this (back when we wanted multiprocessing) had a similar function.

**Comment by ericmjl** [comment]
Great idea! :) What's your idea for implementation?

**Comment by pharos916** [comment]
Since networks require only a bunch of nodes, would it be possible to combine multiple smaller lists of nodes into a larger one? 

1) The data type we use to generate our genotype networks is a dictionary (sequences in protein.py)
2) I was thinking we could transform pickled lists into dictionaries
3) We then combine the multiple dictionaries into a complete one
4) Then, I can use a function similar to generate_genotype_network to use the combined dictionary to synthesize the combined genotype network

Does that make sense? Any comments? Thanks!

**Comment by ericmjl** [comment]
It's a good starting point, Kyle! That said, I think this would not be the optimal implementation/solution.

Suppose you had two networks, one in your left hand, one in your right. Under your solution, all potential nodes get combined into one big list, and the edges are computed from scratch, is this right?

You don't need to compare nodes in the left to those in the left; same goes for the right hand side. To reduce redundant computations, only compare all of the left nodes to all of the right nodes. 

You may wish to add in a few more checks as well:
1. Ensure that the node list on the left side has no overlaps with the node list on the right side. What would you do if there were node overlaps?
2. Anything else?

**Comment by pharos916** [comment]
True, we would only need to compare the left and right nodes. I suppose we could instantiate two separate dictionaries (one for each network), compute only the nodes between the two dicts and not do the computation if the two nodes are the same (check for duplicate at this stage). However, I am not sure how we can implement this solution if we want to combine more than two networks. If we wanted to combine say 4 different networks, that might be difficult. 

Whereas with my first suggestion, combining larger numbers of networks would be straightforward. Combine all the nodes into a pool. I thought about the possibility of duplicate sequences, which are basically the same as node overlaps right? Since the networks I will combine would have already been computed (using generate_genotype_network), the only duplicates would arise from nodes that overlap between our two networks. 

Does all this make sense? Sorry if my answer seems muddled...

**Comment by ericmjl** [comment]
One question I have is this: Why would you instantiate a new dictionary from the network? Wouldn't it save a step to directly iterate over the nodes? I think the part about your proposed implementation that I'm having trouble wrapping my head around is why you would need a dictionary, if the input is a NetworkX object pickled to disk.
